### DISCUSSION TITLE
[Feature] IOS app - limit cache size

### DISCUSSION THREAD

**Discussion by Stooovie** [immich-app/immich#4560] [Feature Request]
### The feature

I'd like to propose a cache size limit feature in the iOS app. I have a library of 55k assets and on my 64GB iPhone, Immich can easily cache like 8 GB or more. I don't mind slower displaying of the images when browsing. 

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by apomortsev** [comment]
Additionally, please consider adding the manual "clear cache" UI option.

**Reply by voronczoff** [reply]
My cache is 40 Gb now. So to clean it, the only way is to delete app, reinstall, configure from scratch?

**Comment by IronButterfly** [comment]
Please add the ability to clear a huge cache in the iOS app. Some kind of button "Clean cache" I mean.

**Comment by dinccey** [comment]
Same goes for the Android app!

**Comment by SapphicCode** [comment]
+1, this is quite wasteful and disrespectful to other apps.

![IMG_2914](https://github.com/user-attachments/assets/554e4bfc-247d-4620-8344-b8c3fd218d11)

**Reply by mertalev** [reply]
Did you back up a lot of assets that were in iCloud Photos? I think there's a bug that causes storage usage to be inflated because the app doesn't clean up data correctly in this case. We don't intentionally try to cache nearly this much data of course.

**Reply by Stooovie** [reply]
(I'm the OP but not the person you're asking, but the info may be relevant anyway).

My entire Immich library was uploaded by Immich running on an iPad (currently 18.5), syncing its iCloud Photo Library to it. Right now, Immich storage use on that iPad sits at 1.83 GB, so the problem is not there anymore for me. On an iPhone that acts as an Immich browser only (syncing is completely disabled in Immich on it), it only uses 200 megs. Everything is up to date as of June 2 2025)

**Reply by thomasss02** [reply]
Android app has the same problem

![Screenshot_2025-06-02-10-59-53-33_fc704e6b13c4fb26bf5e411f75da84f2.jpg](https://github.com/user-attachments/assets/64697e65-889c-4f5f-a3e5-4f0fdc738e88)

**Reply by mertalev** [reply]
Thanks, we should look into this. I suspect this might be another cleanup issue, potentially if there's an error at some point before the files are cleaned up.

**Reply by SapphicCode** [reply]
> Did you back up a lot of assets that were in iCloud Photos? I think there's a bug that causes storage usage to be inflated because the app doesn't clean up data correctly in this case. We don't intentionally try to cache nearly this much data of course.

No backup was enabled on this install. Just scrolling around the main timeline.

**Reply by WuSiYu** [reply]
> Did you back up a lot of assets that were in iCloud Photos? I think there's a bug that causes storage usage to be inflated because the app doesn't clean up data correctly in this case. We don't intentionally try to cache nearly this much data of course.

In my case I'am backup many photos on iCloud Photos, and the immich takes lot of spaces (30G+)

**Reply by apomortsev** [reply]
Was uploading bunch of assets from Google Photo to Immich mobile app on iPad via the “Share” function in Google Photo. Upload broke mid-way due to iPad storage becoming full. Immich storage according to “iPad storage” in Settings is almost 10GB, clicking “Clear file cache” in the Immich app reports freeing up around 300MB with no actual effect on the multi-gigabyte iPad storage consumption. Offloading the app does not help. I can see no other option or setting to clean the space short of re-installing the app.

**Comment by HassanJbara** [comment]
Guys how is this not solved yet?
